### PR TITLE
Improved LCD support and made some park/homing fixes

### PR DESCRIPTION
**Pull request by ClutchplateDude**
- Added RA Autohoming support to Guided startup
- Added DEC Offset homing support to Guided startup
- Defaulted Auto-home questions to YES on LCD during Guided Startup
- Fixed bug that did not honor TrackingOffOnBoot flag when using Guided startup

- Added DEC Offset homing support to CTRL menu
- Removed SetParkingPos from CAL menu
- Added SetDecHomingOffset to CAL menu
- Fixed Park command to actually park DEC after homing
- Fixed a bug that made DEC parking go in the wrong direction
- Made Unpark go to home position after enabling Tracking

- Fixed SetLimit commands to actually use passed limits
- Fixed bug that did not change submenu items in the CTRL menu as button up and down were pressed
- Renamed some menu items for clarity and consistency

### CODE REVIEW

File: Version.h [hunk 1]
```diff
@@ -3,4 +3,4 @@
 // Also, numbers are interpreted as simple numbers.                        _   __   _
 // So 1.8 is actually 1.08, meaning that 1.12 is a later version than 1.8.  \_(..)_/
 
-#define VERSION "V1.11.9"
+#define VERSION "V1.13.1"
```
Comment (julianneswinoga) [inline Version.h]: Could you explain the version bump? 13 vs 12, and why a minor and not just patch?

File: src/Mount.cpp [hunk 2]
```diff
@@ -3306,13 +3322,20 @@ void Mount::moveSteppersTo(float targetRASteps, float targetDECSteps)
...
+        }
         targetDECSteps = min(targetDECSteps, (float) _decUpperLimit);
-        LOG(DEBUG_STEPPERS, "[STEPPERS]: MoveSteppersTo: DEC Upper Limit enforced. To: %f", targetDECSteps);
     }
+
     if (_decLowerLimit != 0)
     {
+        if (targetDECSteps < (float) _decLowerLimit)
+        {
+            LOG(DEBUG_STEPPERS, "[STEPPERS]: MoveSteppersTo: DEC Lower Limit enforced. To: %f", _decLowerLimit);
```
Comment (julianneswinoga) [inline src/Mount.cpp]: `%f` should be `%l`

File: src/c65_startup.hpp [hunk 3]
```diff
@@ -57,28 +69,68 @@ bool processStartupKeys()
     bool waitForRelease = false;
     switch (startupState)
```
Comment (julianneswinoga) [inline src/c65_startup.hpp:70]: Just want to double check that you've tested the LCD with gyro level and auto-home. (I know you usually don't have an LCD)

**Review by julianneswinoga — DISMISSED** [review]

File: Version.h (hunk 1 above)
Comment (ClutchplateDude) [reply Version.h]: I had 12 in another branch, but I guess I could go with 12 here. Given that it has some new features I thought a minor number change was appropriate

File: src/Mount.cpp (hunk 2 above)
Comment (ClutchplateDude) [reply src/Mount.cpp]: Oooh, good catch!

File: src/c65_startup.hpp (hunk 3 above)
Comment (ClutchplateDude) [reply src/c65_startup.hpp:70]: Actually, that's one combo I have not tested. Will do so today.

File: Version.h (hunk 1 above)
Comment (julianneswinoga) [reply Version.h]: Yeah I agree, large change but just wanted to make sure we were thinking the same thing. Minor version 12 makes sense. Keep it in order

File: Version.h [hunk 4]
```diff
@@ -3,4 +3,4 @@
 // Also, numbers are interpreted as simple numbers.                        _   __   _
 // So 1.8 is actually 1.08, meaning that 1.12 is a later version than 1.8.  \_(..)_/
 
-#define VERSION "V1.11.9"
+#define VERSION "V1.12.1"
```
Comment (julianneswinoga) [inline Version.h]: Looking at our previous version bumps, this should be 1.12.0 (not 12.1)

**Review by julianneswinoga — CHANGES_REQUESTED** [review]

File: src/c65_startup.hpp (hunk 3 above)
Comment (ClutchplateDude) [reply src/c65_startup.hpp:70]: I can't rewire my gyro level....

Comment (julianneswinoga) [reply src/c65_startup.hpp:70]: If we're making changes here but can't test them, I'm not sure we should be making those changes

**Review by julianneswinoga — CHANGES_REQUESTED** [review]
Just need someone to test the LCD + gyro combination if we want those changes. Perhaps someone from the discord community would be willing?

File: src/c65_startup.hpp (hunk 3 above)
Comment (ClutchplateDude) [reply src/c65_startup.hpp:70]: So basically stop development on LCD code?

File: src/Mount.cpp [hunk 5]
```diff
@@ -1299,14 +1301,17 @@ void Mount::startSlewingToTarget()
...
     if ((_stepperRA->distanceToGo() != 0) || (_stepperDEC->distanceToGo() != 0))
     {
-        // Only stop tracking if we're actually going to slew somewhere else, otherwise the
+        // Only stop tracking if we're tracking and actually going to slew somewhere else, otherwise the
         // mount::loop() code won't detect the end of the slewing operation...
-        LOG(DEBUG_STEPPERS, "[MOUNT]: Stop tracking (NEMA steppers)");
-        stopSlewing(TRACKING);
-        _trackerStoppedAt        = millis();
-        _compensateForTrackerOff = true;
+        if (slewStatus() == SLEWING_TRACKING)
```
Comment (andre-stefanov) [inline src/Mount.cpp:1367]: `isSlewingTRK()` ?

**Review by andre-stefanov — CHANGES_REQUESTED** [review]

File: src/Mount.cpp (hunk 5 above)
Comment (ClutchplateDude) [reply src/Mount.cpp:1367]: Yes

**Pull request closed without merging**